### PR TITLE
fix: allow env.valueFrom on Terraform

### LINKED ISSUE #8: support env.valueFrom for TF_VAR_ values from Secrets?
being able to create [environment variables from secrets is super handy in k8s](https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets-as-environment-variables) - I wonder if we should extend the [EnvVar](https://github.com/isaaguilar/terraform-operator/blob/master/pkg/apis/tf/v1alpha1/terraform_types.go#L46) struct to also support k8s style `valueFrom` too?

So you could populate your TF vars as:

```yaml
apiVersion: tf.isaaguilar.com/v1alpha1
kind: Terraform
metadata:
  name: tf1
spec:
  env:
  - name: TF_VAR_foo
    value: someValue
  - name: TF_VAR_bar
     valueFrom:
       secretKeyRef:
         name: mysecret
         key: token
```

or is there a nicer TF way to inject secret values?

### PR DESCRIPTION
**Pull request by jstrachan**
so that we can populate terraform values from secrets without putting secrets into the Terraform resource directly

fixes #8

### CODE REVIEW

**Comment by jstrachan** [comment]
note I've not had chance to test this yet, as it requires me to modify the CRD to allow this new syntax - but I hope to get chance to double check it works tomorrow...

File: pkg/apis/tf/v1alpha1/terraform_types.go [hunk 1]
```diff
@@ -177,6 +178,53 @@ type ReconcileTerraformDeployment struct {
 type EnvVar struct {
 	Name  string `json:"name"`
 	Value string `json:"value"`
```
Comment (isaaguilar) [inline pkg/apis/tf/v1alpha1/terraform_types.go]: Will require `omitempty` to make `value` optional.

**Review by isaaguilar — COMMENTED** [review]
I'm wondering if it would be possible to replace the EnvVar with v1.EnvVarSource altogether. 

On a related note, there are some decisions around why EnvVar was a simple key/value and I think it was for aggregating tfvars into a single file. I'll take a closer look at this soon.

**Comment by jstrachan** [comment]
> I'm wondering if it would be possible to replace the EnvVar with v1.EnvVarSource altogether.

yeah - that was my first thought actually; but I was worried it might cause issues with operator-sdk's schema generation. Let me try it though; doing that would be much cleaner

**Comment by isaaguilar** [comment]
@jstrachan Please check your fork, I proposed some changes already.

**Comment by jstrachan** [comment]
@isaaguilar BTW I tried switching to corev1.EnvVar and tested it all out and it worked perfectly!

**Comment by jstrachan** [comment]
@isaaguilar ah whoops - I tried to switch to `corev1.EnvVar` before I noticed your proposed changes - looks like we did the same thing I think ;)

**Comment by jstrachan** [comment]
@isaaguilar ok I switched back to your version of the `corev1.EnvVar` change as you'd also tidied up handling of env vars a bit ;).

I've also tested out your PR too and its working great!

**Pull request merged**